core/src/main/java/org/polypheny/db/processing/Processor.java

```diff
@@ -18,63 +18,61 @@
 import org.polypheny.db.routing.ExecutionTimeMonitor;
 import org.polypheny.db.transaction.Statement;
 import org.polypheny.db.transaction.Transaction;
 import org.polypheny.db.transaction.TransactionException;
 import org.polypheny.db.util.DeadlockException;
 import org.polypheny.db.util.Pair;
 
-
+@Slf4j
 public abstract class Processor {
 
     public abstract List<? extends Node> parse( String query );
 
     public abstract Pair<Node, AlgDataType> validate( Transaction transaction, Node parsed, boolean addDefaultValues );
 
-    public abstract AlgRoot translate( Statement statement, Node query, QueryParameters parameters );
-
-
-    public PolyImplementation prepareDdl( Statement statement, Node parsed, QueryParameters parameters ) {
-        if ( parsed instanceof ExecutableStatement ) {
-            try {
-                // Acquire global schema lock
-                lock( statement );
-                // Execute statement
-                return getResult( statement, parsed, parameters );
-            } catch ( DeadlockException e ) {
-                throw new RuntimeException( "Exception while acquiring global schema lock", e );
-            } catch ( TransactionException | NoTablePrimaryKeyException e ) {
-                throw new RuntimeException( e );
-            } finally {
-                // Release lock
-                unlock( statement );
-            }
-        } else {
-            throw new RuntimeException( "All DDL queries should be of a type that inherits ExecutableStatement. But this one is of type " + parsed.getClass() );
+    public abstract AlgRoot translate( Statement statement, ParsedQueryContext context );
+
+
+    public PolyImplementation prepareDdl( Statement statement, ExecutableStatement node, ParsedQueryContext context ) {
+        try {
+            // Acquire global schema lock
+            lock( statement );
+            // Execute statement
+            return getImplementation( statement, node, context );
+        } catch ( DeadlockException e ) {
+            throw new GenericRuntimeException( "Exception while acquiring global schema lock", e );
+        } catch ( TransactionException e ) {
+            throw new GenericRuntimeException( e );
+        } finally {
+            // Release lock
+            unlock( statement );
         }
+
     }
 
 
-    PolyImplementation getResult( Statement statement, Node parsed, QueryParameters parameters ) throws TransactionException, NoTablePrimaryKeyException {
-        ((ExecutableStatement) parsed).execute( statement.getPrepareContext(), statement, parameters );
+    PolyImplementation getImplementation( Statement statement, ExecutableStatement node, ParsedQueryContext context ) throws TransactionException {
+        node.execute( statement.getPrepareContext(), statement, context );
         statement.getTransaction().commit();
-        Catalog.getInstance().commit();
+        //Catalog.getInstance().commit();
```

Comment:
Why not committing the catalog?

Reply:
It is already committed in the transaction with the statement on the previous line, this is an artifact and would commit twice. Removed.